File: Atrium/src/main/java/ro/upt/atrium/controller/CourseDTO.java
package ro.upt.atrium.controller;

import ro.upt.atrium.model.Professor;
import ro.upt.atrium.model.Slot;
import ro.upt.atrium.model.Student;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

public record CourseDTO(String courseName, String algorithm,
                        LocalDate preferencesDeadline, Professor professor,
                        List<Slot> slots, List<Student> students,
                        boolean finalized, Map<Long, List<Student>> groups) {

    @Override
    public String toString() {
        return "CourseDTO{" +
                "courseName='" + courseName + '\'' +
                ", algorithm='" + algorithm + '\'' +
                ", preferencesDeadline=" + preferencesDeadline +
                ", professor=" + professor +
                ", slots=" + slots +
                ", students=" + students +
                ", finalized=" + finalized +
                ", groups=" + groups +
                '}';
    }
}

File: atrium_frontend/src/app/classes/coursedto.ts
import { Professor } from "./professor";
import { Slot } from "./slot";
import { Student } from "./student";


export class CourseDTO {

    courseid?: number | undefined 
    
    algorithm: string | undefined;
    courseName: string | undefined;
    preferencesDeadline: Date | undefined;
    slots: Slot[] | undefined;

    professor: Professor | undefined;
    finalized: boolean | undefined;
    groups: Map<number, Student[]> | undefined;
}